src/views/MetocDataView.vue

```diff
@@ -660,6 +670,32 @@ export default class MetocDataView extends Mixins(WMSMixin, TimeSeriesMixin) {
     }
   }
 
+  beforeRouteUpdate(to: Route, from: Route, next: NavigationGuardNext) {
+    const goingToLocationRoute = to.params.locationId !== '' && to.params.locationId !== undefined
+    const goingToCoordRoute = to.params.xCoord !== '' && to.params.xCoord !== undefined
+    const sourceIdIsTheSame = to.params.dataSourceId === from.params.dataSourceId
+
+    if (goingToCoordRoute || goingToLocationRoute || sourceIdIsTheSame) {
+      next()
+      return
+    }
+
+    const locationId = from.params.locationId
+    const hasLocation = locationId !== '' && locationId !== undefined
```

Comment:
Misschien voor de consistentie `comingFromLocationRoute`? Je moet nu wel even kijken om te zien of dit om `to` of `from` gaat.